### PR TITLE
Adding Volumetric Lights post-process into Standard Rendering Pipeline

### PR DESCRIPTION
**Pull request by julien-moreau**
Adding "Position Buffer" to geometry renderer as optional
Fixed BloomEnabled in Standard pipeline

### CODE REVIEW

File: src/Shaders/standard.fragment.fx [hunk 1]
```diff
@@ -112,6 +112,82 @@ void main(void)
 }
 #endif
 
+#if defined(VLS)
+#define NB_STEPS 50.0
```
Comment (deltakosh) [inline src/Shaders/standard.fragment.fx]: this should be a parameter no?

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts [hunk 2]
```diff
@@ -208,10 +224,39 @@
             return this._hdrEnabled;
         }
 
+        @serialize()
+        public get VLSEnabled(): boolean {
+            return this._vlsEnabled;
+        }
+
+        public set VLSEnabled(enabled) {
+            var geometry = this._scene.enableGeometryBufferRenderer();
```
Comment (deltakosh) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: you should test if this._vlsEnabled === enabled and return

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts [hunk 3]
```diff
@@ -208,10 +224,39 @@
...
+        }
+
+        public set VLSEnabled(enabled) {
+            var geometry = this._scene.enableGeometryBufferRenderer();
+            if (!geometry.isSupported) {
+                Tools.Warn("Geometry renderer is not supported, cannot create volumetric lights in Standard Rendering Pipeline");
+                return;
+            }
+
+            if (enabled && !this._vlsEnabled) {
```
Comment (deltakosh) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: and then this will be simple

Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: Working on it!

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts [hunk 4]
```diff
@@ -282,6 +327,13 @@
             this.textureAdderFinalPostProcess = new PostProcess("HDRDepthOfFieldSource", "standard", [], [], ratio, null, Texture.BILINEAR_SAMPLINGMODE, scene.getEngine(), false, "#define PASS_POST_PROCESS", Engine.TEXTURETYPE_UNSIGNED_INT);
             this.addEffect(new PostProcessRenderEffect(scene.getEngine(), "HDRBaseDepthOfFieldSource", () => { return this.textureAdderFinalPostProcess; }, true));
 
+            // Create volumetric light
```
Comment (deltakosh) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: I'm wondering if postprocesses should be created ONLY if required (on first call?)

File: src/Shaders/standard.fragment.fx (hunk 1 above)
Comment (julien-moreau) [inline src/Shaders/standard.fragment.fx]: Adding as a parameter

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts (hunk 2 above)
Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: Good catch!

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts (hunk 4 above)
Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: Then it may modify the overall pipeline code but I can do it

Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: I'll do like the default rendering pipeline

Comment (deltakosh) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: yes this way it will create only when required 
Thanks :)

Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: Polish time, I forgot to add the new post-processes to dispose

File: src/Shaders/standard.fragment.fx (hunk 1 above)
Comment (julien-moreau) [inline src/Shaders/standard.fragment.fx]: DONE

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts (hunk 4 above)
Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: DONE

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts (hunk 3 above)
Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: DONE

File: src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts (hunk 2 above)
Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: On each setter ?

Comment (julien-moreau) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.standardRenderingPipeline.ts]: DONE

**Comment by julien-moreau** [comment]
I pushed my refactored version

**Comment by CraigFeldspar** [comment]
Everything seems good to me concerning MRTs !

**Comment by julien-moreau** [comment]
Thanks @CraigFeldspar for reviewing!

**Pull request merged**